toc/working-agreements/2021-11-9-github-teams.md

```diff
@@ -0,0 +1,21 @@
+# Github Teams and Access
+
+## Problem
+We are changing how we operate with working groups and new roles like tech lead and approver. Many (all?) working groups also have sub-groups. Let’s make github teams to reflect these groups.
+
+Teams are used for two purposes:
+* To provide the team members with proper push access to certain repos
+* To provide a way for others to tag groups of people.
+
+## Solution
+
+| Name of Team  | Permissions  |
+|---|---|
+| TOC  |  Admin access to everything. |
+| WG-Leads  |  Write access to community repo. |
+| WG-[WORKING-GROUP-NAME] (this would include approvers and leads)  |  None. Just for organization and tagging. |
+| WG-[WORKING-GROUP-NAME]-Leads  | Admin access for all WG repos.  |
+| WG-[WORKING-GROUP-NAME]-[SUBGROUP-NAME]-Approvers  |  Write access for all repos in the subgroup. |
+| WG-[WORKING-GROUP-NAME]-Bots  | Write access for all WG repos.  |
```

Comment:
```suggestion
| Name of Team  | Team Membership  | Permissions  |
|---|---|---|
| toc | Technical Oversight Committee | Admin access to everything |
| wg-leads | Tech and Execution Leads for all WGs | Write access to community repo |
| wg-[WORKING-GROUP-NAME] | All approvers and leads for a WG | None: only for organization and tagging |
| wg-[WORKING-GROUP-NAME]-leads | All leads for a WG | Admin access for all WG repos |
| wg-[WORKING-GROUP-NAME]-[SUBGROUP-NAME]-approvers | Approvers for a subgroup within a WG | Write access for all repos in the subgroup |
| wg-[WORKING-GROUP-NAME]-bots | Bot accounts for a WG | Write access for all WG repos |
```

* `[WORKING-GROUP-NAME]` is the name of a particular working group, converted to kebab case. 
* `[SUBGROUP-NAME]` is the name of a subgroup within a particular working group, converted to kebab case.

Reply:
the approvers group should also include the tech leads in my opinion. Because that will allow us to use [auto-assignment](https://docs.github.com/en/organizations/organizing-members-into-teams/managing-code-review-settings-for-your-team#about-auto-assignment).

Reply:
Could be done by including all -leads tesms into the corresponding -approvers teams.

Reply:
Do approvers actually need write access? When using Code Review Limits approvers would actually only need explicit read access on the repo: 
<img width="656" alt="Screenshot 2021-11-26 at 20 53 36" src="https://user-images.githubusercontent.com/380697/143929489-50a08971-2442-47cf-be9f-f793f6e845e6.png">

Just thinking about the least permissions required to make things work. We would still need something to merge the actual PRs but maybe that could be an auto-merge bot, once the required number of reviewer approvals has been reached.

Reply:
Isn't setting labels and assigning oneself to issue limited to write access as well? This could also be handled by by a bot reacting to comands though. Kubernetes and Istio communities have examples for that as well.

Reply:
Having the wg and area names concatenated will result in really long names: `wg-foundational-infrastructure-system-logging-and-metrics-approvers`.

Does it make sense to just put area teams inside a parent team and not concatenate the names?

Reply:
If we can avoid name clashes by means other than namespacing, I don't see a problem here (if one can easily navigate from child to parent team).

Reply:
What permissions are needed on repos to set up CI integrations? I'm thinking both about "checks" that call out to external concourse/etc deployments, and also github actions for folks who might go that way.